deeppavlov/models/go_bot/tracker/dialogue_state_tracker.py

```diff
@@ -33,28 +34,91 @@ def get_current_knowledge(self) -> DSTKnowledge:
-                                    nlg_manager.num_of_known_actions(), nlg_manager.get_api_call_action_id(),
-                                    policy_network_params.hidden_size,
-                                    database)
+        slot_names = parent_tracker.slot_names
+
+        # region set formfilling info
+        act2act_id = {a_text: nlg_manager.get_action_id(a_text) for a_text in nlg_manager.known_actions()}
+        action_id2aqd_slots_ids, action_id2req_slots_ids = DialogueStateTracker.extract_reqiured_acquired_slots_ids_mapping(
+            act2act_id, slot_names, nlg_manager, parent_tracker)
+
+        # todo why so ugly and duplicated in multiple users tracker
+        dialogue_state_tracker = DialogueStateTracker(slot_names, nlg_manager.num_of_known_actions(),
+                                                      nlg_manager.get_api_call_action_id(),
+                                                      policy_network_params.hidden_size,
+                                                      database,
+                                                      parent_tracker.domain_yml_path,
+                                                      parent_tracker.stories_path)
+
+        dialogue_state_tracker.ffill_act_ids2req_slots_ids = action_id2req_slots_ids
+        dialogue_state_tracker.ffill_act_ids2aqd_slots_ids = action_id2aqd_slots_ids
+
+        # endregion set formfilling info
+        return dialogue_state_tracker
+
+    @staticmethod
+    def extract_reqiured_acquired_slots_ids_mapping(act2act_id: Dict,
+                                                    slot_names: List,
+                                                    nlg_manager: NLGManagerInterface,
+                                                    parent_tracker: FeaturizedTracker) -> Tuple[Dict[str, np.ndarray], Dict[str, np.ndarray]]:
+        """
+        get the required and acquired slots information for each known action in the -Hot Encoding form
+        Args:
+            act2act_id: the mapping of actions onto their ids
+            slot_names: the names of slots known to the tracker
+            nlg_manager: the NLG manager used in system
+            parent_tracker: the tracker to take required and acquired slots information from
+
+        Returns:
+            the dicts providing np.array masks of required and acquired slots for each known action
+        """
+        action_id2aqd_slots_ids = dict()  # aqd stands for acquired
+        action_id2req_slots_ids = dict()
+        for act in nlg_manager.known_actions():
+            act_id = act2act_id[act]
+
+            action_id2req_slots_ids[act_id] = np.zeros(len(slot_names), dtype=np.float32)
+            action_id2aqd_slots_ids[act_id] = np.zeros(len(slot_names), dtype=np.float32)
+
+            if isinstance(act, tuple):
+                acts = act
+            else:
+                acts = [act]
+
+            for act in acts:
```

Comment:
Variable 'act' already declared in 'for' loop statement above. Not a mistake but not a best practice either